inception-ui-external-search/src/main/java/de/tudarmstadt/ukp/inception/app/ui/externalsearch/sidebar/ExternalSearchAnnotationSidebar.java

```diff
@@ -0,0 +1,427 @@
+        extends Panel
+    {
+        public ResultRowView(String id, long rowNumber, IModel<ExternalSearchResult> model)
+        {
+            super(id, model);
+
+            ExternalSearchResult result = (ExternalSearchResult) getDefaultModelObject();
+
+            String documentTitle = result.getDocumentTitle();
+
+            Whitelist wl = new Whitelist();
+            wl.addTags("em");
+            Document dirty = Jsoup.parseBodyFragment(result.getHighlights().get(0), "");
+            Cleaner cleaner = new Cleaner(wl);
+            Document clean = cleaner.clean(dirty);
+            clean.select("em").tagName("mark");
+            String highlight = clean.body().html();
+
+            boolean existsSourceDocument = documentService
+                .existsSourceDocument(project, documentTitle);
+
+            // Import and open annotation
+            LambdaAjaxLink link;
+            if (!existsSourceDocument) {
+                link = new LambdaAjaxLink("docLink", _target -> {
+                    selectedResult = result;
+                    try {
+                        documentImporter
+                            .importDocumentFromDocumentRepository(result.getDocumentTitle(),
+                                searchStateModel.getObject().getCurrentRepository());
+
+                        getAnnotationPage().actionShowSelectedDocument(_target,
+                            documentService.getSourceDocument(project, documentTitle));
+                    }
+                    catch (IOException e) {
+                        LOG.error(e.getMessage(), e);
+                        error(e.getMessage() + " - " + ExceptionUtils.getRootCauseMessage(e));
+                        e.printStackTrace();
+                    }
+                });
+            }
+            else {
+                // open action
+                link = new LambdaAjaxLink("docLink", _target -> {
+                    selectedResult = result;
+                    getAnnotationPage().actionShowSelectedDocument(_target,
+                        documentService.getSourceDocument(project, documentTitle));
+                });
+            }
+
+            String title = defaultIfBlank(result.getDocumentTitle(),
+                defaultIfBlank(result.getDocumentId(),
+                    defaultIfBlank(result.getUri(), "<no title>")));
+
+            add(link);
+
+            link.add(new Label("title", title));
+            link.add(new Label("score", result.getScore()));
+            link.add(new Label("highlight", highlight).setEscapeModelStrings(false));
+            link.add(new Label("importStatus",
+                () -> existsSourceDocument ? "imported" : "not imported"));
+        }
+    }
+
+    public static class ExternalSearchUserState
+        implements Serializable
+    {
+        private AnnotationLayer layer;
+
+        private DocumentRepository currentRepository = null;
+
+        private IModel<String> targetQuery = null;
```

Comment:
It should be ok if this is a `String` since the entire `ExternalSearchUserState` is already acting as a model.

Reply:
The model is still being used by the search form
`add(new TextField<>("queryInput", searchStateModel.getObject().getQuery(), String.class));`